File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/FetchMessages.java
package com.pubnub.api.java.endpoints;

import com.pubnub.api.models.consumer.history.PNFetchMessagesResult;

public interface FetchMessages extends Endpoint<PNFetchMessagesResult> {
    FetchMessages channels(java.util.List<String> channels);

    FetchMessages maximumPerChannel(Integer maximumPerChannel);

    FetchMessages start(Long start);

    FetchMessages end(Long end);

    FetchMessages includeMeta(boolean includeMeta);

    FetchMessages includeMessageActions(boolean includeMessageActions);

    FetchMessages includeMessageType(boolean includeMessageType);

    FetchMessages includeUUID(boolean includeUUID);

    FetchMessages includeCustomMessageType(boolean includeCustomMessageType);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/FetchMessages.kt
package com.pubnub.api.endpoints

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.PNBoundedPage
import com.pubnub.api.models.consumer.history.PNFetchMessagesResult

/**
 * @see [PubNub.fetchMessages]
 */
actual interface FetchMessages : Endpoint<PNFetchMessagesResult> {
    val channels: List<String>
    val page: PNBoundedPage
    val includeUUID: Boolean
    val includeMeta: Boolean
    val includeMessageActions: Boolean
    val includeMessageType: Boolean
    val includeCustomMessageType: Boolean
}
